react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  useEffect(() => {
+    const possibleItems = readItems();
+    possibleItems.then(
+      (value) => {
+        setItems(value);
+      },
+      (error) => {
+        setError(error);
+      }
+    );
+    setIsLoading(false);
```

Comment:
This actually doesn't work like you want it to. It will set isLoading to false _before_ the readItems Promise completes.

Trace the execution:
Line 15 - call readItems and get a Promise
Line 16 - set up the .then handler on the returned Promise
Line 17-23 - these don't execute until the Promise fulfills
Line 24 - executes immediately after Line 16 and sets isLoading false even though the readItems Promise has not yet fulfilled.